app/models/server.rb

```diff
@@ -276,13 +280,27 @@ def self.find(id)
   end
 
   # Find the server with the lowest load (for creating a new meeting)
-  def self.find_available
+  def self.find_available(tag_arg = nil)
+    # Check if tag is required
+    tag = tag_arg
+    tag_required = false
+    if !tag.nil? && tag[-1] == '!'
+        tag = tag[0..-2] # always returns String, if tag is String
+        tag_required = true
+    end
```

Comment:
The block above works as expected for tag_arg = any string with length > 1 and also nil, "", "!".